### PR TITLE
Resolve out of range error

### LINKED ISSUE #80: Kraft failes with list index out of range
**Describe the bug**
Right now if you are trying to run the **hello-world** application you will end up with **list index out of range** error. This can be for all applications if the name of the directory of the application is different from the name of the application from **kraft.yml**.

**To Reproduce**
1. git clone git@github.com:unikraft/app-helloworld.git
2. cd app-helloworld
3. kraft configure -p kvm -m x86_64
4. kraft build
5. kraft run

**Expected behavior**
The error is not descriptive at all, it is more of a python error than something suggestive so you can figure out what to do with kraft. From my point of view, it would be better to suggest to you that kraft doesn't find the executable.

**Screenshots**
Screenshot from 2021-10-30 10-38-11

**Desktop (please complete the following information):**
 - OS: Ubuntu

**Additional context**

### PR DESCRIPTION
**Pull request by gabrielmocanu**
Add a check for `out of range` error if the name of the base directory is different from the name of the application from kraft.yaml.

Github-Pull-Request-Closes: #80 
Signed-off-by: Gabriel Mocanu <gabi.mocanu98@gmail.com>

### CODE REVIEW

File: kraft/cmd/run.py [hunk 1]
```diff
@@ -88,6 +88,9 @@ def kraft_run(ctx, appdir=None, target=None, plat=None, arch=None, initrd=None,
 
             binaries.append(binname)
 
+        if(len(binaries) == 0):
```
Comment (razvand) [inline kraft/cmd/run.py]:
```suggestion
        if (len(binaries) == 0):
```

File: kraft/cmd/run.py [hunk 2]
```diff
@@ -88,6 +88,9 @@ def kraft_run(ctx, appdir=None, target=None, plat=None, arch=None, initrd=None,
...
+            raise ValueError("Unikernel binary not found.(Please check that your application name is the same as directory name)")
```
Comment (razvand) [inline kraft/cmd/run.py]:
```suggestion
            raise ValueError("Unikernel binary not found. Please check that your application name (in the kraft.yaml file) is the same as directory name.")
```

**Review by razvand — CHANGES_REQUESTED** [review]
I made some layout / wording suggestions.

**Review by StefanJum — COMMENTED** [review]
When using `kraft up -t helloworld different-name-dir` this does not work, since the `if` on [this line](https://github.com/unikraft/kraft/blob/staging/kraft/cmd/run.py#L79) is not taken, and `target` is set [here](https://github.com/unikraft/kraft/blob/staging/kraft/cmd/run.py#L75).
Works fine with `kraft run`.

**Comment by gabrielmocanu** [comment]
I closed this PR since this [one] fixed the issue.

[one]: https://github.com/unikraft/kraft/pull/94

**Pull request closed without merging**